lib/src/Uncertainty/Algorithm/MetaModel/TensorApproximation/CanonicalTensorEvaluation.cxx

```diff
@@ -0,0 +1,243 @@
+//                                               -*- C++ -*-
+/**
+ *  @brief Canonical tensor representation and evaluation
+ *
+ *  Copyright 2005-2017 Airbus-EDF-IMACS-Phimeca
+ *
+ *  This library is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+#include "openturns/CanonicalTensorEvaluation.hxx"
+#include "openturns/PersistentObjectFactory.hxx"
+#include "openturns/UniVariateFunctionEvaluation.hxx"
+
+BEGIN_NAMESPACE_OPENTURNS
+
+CLASSNAMEINIT(CanonicalTensorEvaluation);
+
+static const Factory<CanonicalTensorEvaluation> Factory_CanonicalTensorEvaluation;
+
+
+CanonicalTensorEvaluation::CanonicalTensorEvaluation()
+: NumericalMathEvaluationImplementation()
+{
+  // Nothing to do
+}
+
+
+CanonicalTensorEvaluation::CanonicalTensorEvaluation(const FunctionFamilyCollection & functionFamilies,
+                                                     const Indices & nk,
+                                                     const UnsignedInteger rank)
+: NumericalMathEvaluationImplementation()
+, functionFamilies_(functionFamilies)
+, degrees_(nk)
+, coefficients_(nk.getSize())
```

Comment:
It's a collection of Samples, the use-case is more to initialize them with the right size/dim than knowing the coefficients in advance.